GitUI/GitUIExtensions.cs

```diff
@@ -87,11 +89,32 @@ public static Task ViewChanges(this FileViewer diffViewer, IList<GitRevision> re
 
         public static Task ViewChanges(this FileViewer diffViewer, string firstRevision, string secondRevision, GitItemStatus file, string defaultText)
         {
-            return diffViewer.ViewPatch(() =>
+            if (firstRevision == null)
+            {
+                // The previous commit does not exist, nothing to compare with
+                if (file.TreeGuid.IsNullOrEmpty())
+                {
+                    if (secondRevision.IsNullOrWhiteSpace())
+                    {
+                        // Revision unexpectedly is not set, just return for 2.51 update
+                        return Task.CompletedTask;
```

Comment:
I don't understand why you want to silently swallow this. It should never happen, however if some path leads to this case I just  would like to throw an exception here and get a bug report about it happening. Otherwise an user can be misled about the diff content.

Reply:
The diff will be empty, not incorrect.
I propose to leave the issue report to master. This would normally occur in a transient situations when switching revisions, or something. The users sees this as a blank diff so if there are situations where this occurs in valid situations, there will be visible anyway.

Reply:
>The diff will be empty, not incorrect.

It is hard to say not knowing circumstances at which it happened. 
In your solution you can at best get a bug report complaining about an empty diff. In my solution it is likely you get a call stack pointing to the bad invocation place.

Reply:
@gerhardol let's throw